### PR TITLE
fix(mobile): omit null values from json requests

### PR DESCRIPTION
**Pull request by jrasm91**
This PR generates the dart sdk with a custom template, that omits null values from the generated json object.

### CODE REVIEW

**Comment by zoodyy** [comment]
I really like the idea of omitting null values. My main concern is with the new `native_class.mustache` file. I assume its taken from openapi and modified by hand? If so, we won't see changes made to the openapi default template. Can we instead make a patch file (to modify the default template) and re-generate & store that new template whenever the generator is run?

**Comment by jrasm91** [comment]
Hmm, that's an interesting idea. I don't know how easy that would be to do, but I can take a look.

**Comment by jrasm91** [comment]
@zoodyy I moved the open-api generation into a script, which now downloads the latest `native_class.mustache` file from github, applies the patch and then generates the new files. There were in fact already a few changes to the template that needed to be merged in :smile:.

**Comment by zoodyy** [comment]
@jrasm91 Excellent! Funny indeed that the new approach already found some changes. :laughing: 
I'll take a look at the changes.

File: server/package.json [hunk 1]
```diff
@@ -27,9 +27,7 @@
     "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
     "test:e2e": "jest --config ./apps/immich/test/jest-e2e.json",
     "typeorm": "node --require ts-node/register ./node_modules/typeorm/cli.js",
-    "api:typescript": "rm -rf ../web/src/api/open-api && npx openapi-generator-cli generate -g typescript-axios -i ./immich-openapi-specs.json -o ../web/src/api/open-api",
-    "api:dart": "rm -rf ../mobile/openapi && npx openapi-generator-cli generate -g dart -i ./immich-openapi-specs.json -o ../mobile/openapi",
-    "api:generate": "npm run api:typescript && npm run api:dart"
+    "api:generate": "./bin/generate-open-api.sh"
```
Comment (zoodyy) [inline server/package.json:30]: I'm unsure about this change. On the plus side, always generating mobile+web leaves less room for errors. But it also takes longer and is unnecessary if you are only developing mobile or web at a time.
@alextran1502 Do you like to use the individual API generate commands?

File: server/bin/generate-open-api.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+# mobile
+rm -rf ../mobile/openapi
+
+cd ./openapi-generator/templates/serialization/native
+
+wget -O native_class.mustache https://raw.githubusercontent.com/OpenAPITools/openapi-generator/master/modules/openapi-generator/src/main/resources/dart2/serialization/native/native_class.mustache
```
[the remark this answers is not in the document]
Comment (jrasm91) [reply server/bin/generate-open-api.sh:8]: Agreed, I had the same thought, but they don't have a release branch, so I want sure how to get that specific version.

File: server/package.json (hunk 1 above)
Comment (jrasm91) [reply server/package.json:30]: I currently run `make api`, which does both even though I only develope for web.

I'm happy to make it possible to call them separately if this is inconvenient or takes too long. For me personally it is pretty quick, not to mention I don't actually use it very often. 

On the plus side they would always be in sync this way.

Main motivation was to have all the generator code in one place, but I could add some arg processing to do web, mobile, or both.

**Pull request merged**